contracts/utils/StrikeSelection.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {IOtoken} from "../interfaces/GammaInterface.sol";
+import {DSMath} from "../lib/DSMath.sol";
+import {IOptionsPremiumPricer} from "../interfaces/IRibbon.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StrikeSelection is DSMath, Ownable {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IOptionsPremiumPricer public immutable optionsPremiumPricer;
+
+    // delta for options strike price selection (ex: 0.1 is 1)
+    uint256 public delta;
+    // step pct at which we will iterate over
+    // (ex: 500 = 50% means we will move in leaps of 50% of spot price)
+    uint256 public step;
+
+    event DeltaSet(uint256 oldDelta, uint256 newDelta, address owner);
+    event StepSet(uint256 oldStep, uint256 newStep, address owner);
+
+    constructor(address _optionsPremiumPricer) {
+        require(_optionsPremiumPricer != address(0), "!_optionsPremiumPricer");
+        optionsPremiumPricer = IOptionsPremiumPricer(_optionsPremiumPricer);
+        // set delta to 0.1
+        delta = 1;
+        // set step to 5%
+        step = 50;
+    }
+
+    /**
+     * @notice Gets the strike price satisfying the delta value
+     * given the expiry timestamp and whether option is call or put
+     * @param expiryTimestamp is the unix timestamp of expiration
+     * @param isPut is whether option is put or call
+     */
+
+    function getStrikePrice(uint256 expiryTimestamp, bool isPut)
+        external
+        returns (uint256 strikePrice)
```

Comment:
Would be helpful to return `(strikePrice, delta)` so the consumer can emit the delta value out in an event for transparency

Reply:
👍